src/it/MSHADE-36-inject-dep-reduced-pom-in-final/verify.bsh

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.*;
+import java.util.jar.*;
+import java.util.Arrays;
+import org.codehaus.plexus.util.*;
+
+String[] wanted =
+{
+    "nl/example/Main.class",
+    "junit/swingui/icons/error.gif",
+    "nl/example/shaded/org/codehaus/plexus/util/StringUtils.class",
+};
+
+String[] unwanted =
+{
+    "junit/swingui/TestRunner.class",
+    "org/codehaus/plexus/util/StringUtils.class",
+};
+
+JarFile jarFile = new JarFile( new File( basedir, "target/test-1.0.jar" ) );
+
+for ( String path : wanted )
+{
+    if ( jarFile.getEntry( path ) == null )
+    {
+        throw new IllegalStateException( "wanted path is missing: " + path );
+    }
+}
+
+for ( String path : unwanted )
+{
+    if ( jarFile.getEntry( path ) != null )
+    {
+        throw new IllegalStateException( "unwanted path is present: " + path );
+    }
+}
+
+JarEntry jarEntry = jarFile.getEntry( "META-INF/maven/org.apache.maven.its.shade.drp/test/pom.xml" );
+String pomFile = IOUtil.toString( jarFile.getInputStream( jarEntry ), "UTF-8" );
+
+if ( pomFile.contains( "<groupId>org.codehaus.plexus</groupId>" ) ) {
+    throw new IllegalStateException( "The pom.xml still contains a reference to the org.codehaus.plexus dependency");
+}
+
+jarFile.close();
```

Comment:
Pls call this in `finally {}` block.
It gives at least some guarantee that we do not have open streams when deleting workspace on Jenkins, thx.

Reply:
I've updated the script. Please check if this is what you meant.